src/controller.rs

```diff
@@ -82,9 +82,10 @@ impl FoxBox {
                hostname: Option<String>,
                http_port: u16,
                ws_port: u16,
-               tls_option: TlsOption) -> Self {
-        let profile_service = ProfileService::new(None);
+               tls_option: TlsOption,
+               directory: Option<String>) -> Self {
```

Comment:
Instead of a `Option` we can make `directory` a `enum` like:

```rust
pub enum ProfilePath {
  Default,
  Custom(String)
}
```

and change https://github.com/fxbox/foxbox/blob/d8ccbfa0da2d7c1a0fb74c38fd481210f3216e98/src/profile_service.rs#L27 to use it as well.